File: sentry-samples/kmp-app-mvvm-di/iosApp/LoginScreen.swift
import SwiftUI
import shared

struct LoginScreen: View {
    @State private var email = "user@sentrydemo.com"
    @State private var password = "randompassword"
    @State private var enableLoginError = true
    @State private var showDialog = false
    @State private var dialogMessage = ""
    @State private var isLoggedIn = false
    @Environment(\.presentationMode) var presentationMode

    private var viewModel = KotlinDependencies.shared.getAuthenticationViewModel()
    
    var body: some View {
        VStack(alignment: .center, spacing: 16) {
            Text("Sentry Demo")
                .font(.title)
                .fontWeight(.bold)
                .padding(.bottom, 16)
            TextField("Email", text: $email)
                .keyboardType(.emailAddress)
                .textFieldStyle(RoundedBorderTextFieldStyle())
            SecureField("Password", text: $password)
                .textFieldStyle(RoundedBorderTextFieldStyle())
                .padding(.top, 16)
            Button(action: {
                let succeded = viewModel.login(withError: enableLoginError)
                if succeded {
                    // present home screen
                    presentationMode.wrappedValue.dismiss()
                } else {
                    dialogMessage = "An error occurred during login"
                    showDialog = true
                }
            }) {
                if (enableLoginError) {
                    Text("Log In (error)")
                } else {
                    Text("Log In")
                }
            }
            .frame(maxWidth: .infinity)
            .padding(.top, 16)
            Button(action: {
                viewModel.signUp()
            }) {
                Text("Sign up (crash)")
            }
            .frame(maxWidth: .infinity)
            .padding(.top, 16)
            HStack {
                Text("Enable login error")
                Toggle("", isOn: $enableLoginError)
                    .padding(.trailing, 8)
            }
        }
        .padding(16)
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .alert(isPresented: $showDialog) {
            Alert(
                title: Text("Error"),
                message: Text(dialogMessage),
                dismissButton: .default(Text("OK")) {
                    showDialog = false
                }
            )
        }
    }
}

File: sentry-samples/kmp-app-mvvm-di/androidApp/src/main/kotlin/sentry/kmp/demo/android/ui/LoginScreen.kt
package sentry.kmp.demo.android.ui

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.material.AlertDialog
import androidx.compose.material.Button
import androidx.compose.material.Checkbox
import androidx.compose.material.MaterialTheme
import androidx.compose.material.OutlinedTextField
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.unit.dp
import androidx.navigation.NavController
import sentry.kmp.demo.models.AuthenticationViewModel

@Composable
fun LoginScreen(navController: NavController, authenticationViewModel: AuthenticationViewModel) {
    val email = remember { mutableStateOf("user@sentrydemo.com") }
    val password = remember { mutableStateOf("randompassword") }
    val enableLoginError = remember { mutableStateOf(true) }
    var showDialog by remember { mutableStateOf(false) }
    var dialogMessage by remember { mutableStateOf("") }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(16.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        Text(
            text = "Sentry Demo",
            style = MaterialTheme.typography.h4,
            fontWeight = FontWeight.Bold,
            modifier = Modifier.padding(bottom = 16.dp)
        )
        OutlinedTextField(
            value = email.value,
            onValueChange = { email.value = it },
            label = { Text("Email") },
            keyboardOptions = KeyboardOptions(
                keyboardType = KeyboardType.Email
            ),
            modifier = Modifier.fillMaxWidth()
        )
        OutlinedTextField(
            value = password.value,
            onValueChange = { password.value = it },
            label = { Text("Password") },
            visualTransformation = PasswordVisualTransformation(),
            keyboardOptions = KeyboardOptions(
                keyboardType = KeyboardType.Password
            ),
            modifier = Modifier
                .fillMaxWidth()
                .padding(top = 16.dp)
        )
        Button(
            onClick = {
                val succeeded = authenticationViewModel.login(enableLoginError.value)
                if (succeeded) {
                    navController.navigate("home")
                } else {
                    dialogMessage = "An error occurred during login"
                    showDialog = true
                }
            },
            modifier = Modifier
                .fillMaxWidth()
                .padding(top = 16.dp)
        ) {
            Text("Log In ${if (enableLoginError.value) "(error)" else ""}")
        }
        Button(
            onClick = {
                authenticationViewModel.signUp()
            },
            modifier = Modifier
                .fillMaxWidth()
                .padding(top = 16.dp)
        ) {
            Text("Sign up (crash)")
        }
        Row(
            modifier = Modifier.fillMaxWidth(),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.Start
        ) {
            Text("Enable login error")
            Checkbox(
                checked = enableLoginError.value,
                onCheckedChange = { enableLoginError.value = it },
                modifier = Modifier.padding(end = 8.dp)
            )
        }
        if (showDialog) {
            AlertDialog(
                onDismissRequest = { showDialog = false },
                title = { Text("Error") },
                text = { Text(dialogMessage) },
                confirmButton = {
                    Button(
                        onClick = { showDialog = false }
                    ) {
                        Text("OK")
                    }
                }
            )
        }
    }
}
